controllers/machinehealthcheck_controller_test.go

```diff
@@ -314,9 +315,10 @@ var _ = Describe("MachineHealthCheck", func() {
 					}
 					return &mhc.Status
 				}).Should(Equal(&clusterv1.MachineHealthCheckStatus{
-					ExpectedMachines: 2,
-					CurrentHealthy:   2,
-					Targets:          targetMachines},
+					ExpectedMachines:   2,
+					CurrentHealthy:     2,
+					ObservedGeneration: 1,
```

Comment:
Given the problem of those test, I'm leaning towards to use `Should(MatchMachineHealthCheckStatus(...))` instead of asserting on ` ObservedGeneration: 1`
@vincepri opinions?

Reply:
just checked again, it's the only Equal, agree to change it